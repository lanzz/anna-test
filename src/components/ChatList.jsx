import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { closeChat, selectChat } from '../actions';

function ChatList({ chats, selectedChat, closeChat, selectChat }) {
    const chatsList = chats.map((chat) => (
        <li className={chat === selectedChat? 'selected' : undefined}
            key={chat.id}
            onClick={() => selectChat(chat.id)}
        >
            <button className="close" onClick={() => closeChat(chat.id)}>â•³</button>
            {chat.lastMessage.content}
        </li>
    ));
    return (
        <ul className="chat-list">
            {chatsList}
        </ul>
    );
}

ChatList.propTypes = {
    chats: PropTypes.array.isRequired,
    selectedChat: PropTypes.object,
    closeChat: PropTypes.func.isRequired,
    selectChat: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
    return {
        chats: state.chats,
        selectedChat: state.selectedChat
    };
};

const mapDispatchToProps = { closeChat, selectChat };

export default connect(mapStateToProps, mapDispatchToProps)(ChatList);
