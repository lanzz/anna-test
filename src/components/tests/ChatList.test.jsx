import React from 'react';
import { render } from 'enzyme';
import ChatList from '../ChatList';

it('renders correctly', () => {
    const wrapper = render(
        <ChatList />
    );
    expect(wrapper).toMatchSnapshot();
});
