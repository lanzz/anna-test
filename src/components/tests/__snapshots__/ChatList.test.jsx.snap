// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  class="chat-list"
>
  <li
    class="selected"
  >
    <button
      class="close"
    >
      ╳
    </button>
    You said: hello world
  </li>
  <li>
    <button
      class="close"
    >
      ╳
    </button>
    You said: ok then
  </li>
</ul>
`;
